qfieldsync/core/layer.py

```diff
@@ -153,16 +154,16 @@ def copy(self, target_path):
             # Copy will also be called on non-file layers like WMS. In this case, just do nothing.
             return
 
-        # Shapefiles... have the path in the source
-        file_path = self.layer.source()
+        # Shapefiles and GeoPackages have the path in the source
+        file_path = self.layer.source().split('|')[0]
         # Spatialite have the path in the table part of the uri
         uri = QgsDataSourceUri(self.layer.dataProvider().dataSourceUri())
 
         if os.path.isfile(file_path):
             source_path, file_name = os.path.split(file_path)
             basename, extensions = get_file_extension_group(file_name)
             for ext in extensions:
-                if os.path.exists(os.path.join(source_path, basename + ext)):
+                if os.path.exists(os.path.join(source_path, basename + ext)) and not os.path.exists(os.path.join(target_path, basename + ext)):
```

Comment:
With this change it's no longer possible to export into the same directory than a previous export and overwrite data files.
I'm not sure if that is an issue somewhere, but it may be better to keep track of already copied layers by passing a list of already copied files into this method and check against that list